File: src/main/java/org/example/kalkulationsprogramm/domain/LieferantNotiz.java
package org.example.kalkulationsprogramm.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name = "lieferant_notiz")
public class LieferantNotiz {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lieferant_id", nullable = false)
    private Lieferanten lieferant;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String text;

    @Column(nullable = false)
    private LocalDateTime erstelltAm;

    @PrePersist
    protected void onCreate() {
        if (erstelltAm == null) {
            erstelltAm = LocalDateTime.now();
        }
    }
}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/LieferantNotiz.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "lieferant_notiz")
open class LieferantNotiz {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lieferant_id", nullable = false)
    open var lieferant: Lieferanten? = null

    @Column(columnDefinition = "TEXT", nullable = false)
    open var text: String? = null

    @Column(nullable = false)
    open var erstelltAm: LocalDateTime? = null

    @PrePersist
    open fun onCreate() {
        if (erstelltAm == null) erstelltAm = LocalDateTime.now()
    }

}